src/main/java/com/google/sps/servlets/ManageListServlet.java

```diff
@@ -55,31 +59,58 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String type = request.getParameter("type").toLowerCase();
     String action = request.getParameter("action").toLowerCase();
 
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
     // creates a key based on the passed in recipe ID
     Key key = KeyFactory.createKey("Recipe", recipeID);
     // creates a user based on the passed in user id token
     User user = new User(userIdToken);
 
-    // adds created key to appropriate list based on passed in list type
+    // adds or removes created key to/from appropriate list based on passed in 
+    // list type and passed in action
     if (type.equals("cookbook")) {
-      actionOnPlanner(user, key, action);
-    } else if (type.equals("planner")) {
       actionOnCookbook(user, key, action);
+    } else if (type.equals("planner")) {
+      actionOnPlanner(user, key, action);
     } else {
       System.out.println("AddToListServlet: invalid type " + type);
     }
   }
 
-  public void actionOnPlanner(User user, Key key, String action) {
+  public void actionOnCookbook(User user, Key key, String action) {
     if (action.equals("add")) {
       user.addCookbookKey(key);
+      updatePopularity(key, "increase");
```

Comment:
You should make "increase" and "decrease" be constants (even though they're referenced only twice, it'll keep it cleaner).

That being said if you wanted to skip the actions and the string checks, you could have the second param be a bool named `increase` where true -> popularity++ and false -> popularity-- (or something similar). Up to you though!

Reply:
cool followed the second suggestion!